src/general/system.jl

```diff
@@ -151,3 +151,17 @@ end
 
 # Only for systems requiring a mandatory callback
 reset_callback_flag!(system) = system
+
+# Assuming a constant particle spacing one can calculate the number of neighbors within the
+# compact support for an undisturbed particle distribution.
+function ideal_neighbor_count(::Val{D}, particle_spacing, compact_support) where {D}
+    throw(ArgumentError("Unsupported dimension: $D"))
+end
+
+@inline function ideal_neighbor_count(::Val{2}, particle_spacing, compact_support)
+    return floor(Int, pi * compact_support^2 / particle_spacing^2)
+end
+
+@inline @fastpow function ideal_neighbor_count(::Val{3}, particle_spacing, compact_support)
```

Comment:
Please check the codecov report. This is not covered, but the 2D version is.

Reply:
A test for this exists in #584

Reply:
Can you add it here or does it require other code from #584?